src/account/AccountItem.js

```diff
@@ -59,23 +66,55 @@ export default function AccountItem(props: Props): Node {
 
   const backgroundItemColor = isLoggedIn ? 'hsla(177, 70%, 47%, 0.1)' : 'hsla(0,0%,50%,0.1)';
   const textColor = isLoggedIn ? BRAND_COLOR : 'gray';
+  const accountItemHeight = isLoggedIn ? 88 : 94;
+
+  const [orgDetail, setOrgDetail] = useState({
+    realm_icon: '',
+    realm_name: '',
+    realm_uri: '',
+  });
+
+  useEffect(() => {
+    async function getAndSetIsAvailable() {
+      try {
+        const serverSettings: ApiResponseServerSettings = await api.getServerSettings(realm);
+        setOrgDetail(serverSettings);
+      } catch (e) {
+        showErrorAlert(_('Failed to connect to server: {realm}', { realm: realm.toString() }));
+      }
+    }
+    getAndSetIsAvailable();
+  }, [realm, _]);
+
+  if (!orgDetail.realm_icon || !orgDetail.realm_name || !orgDetail.realm_uri) {
+    return <LoadingIndicator size={40} />;
+  }
```

Comment:
No, don't do this. Remember that people aren't always on a fast Internet connection, or indeed any connection. It's very frustrating to be trying to navigate around an app, looking at information it should already have -- like the list of accounts you're logged into -- and have it sit there hanging while it tries to talk to a server.

If the only way to implement this feature of showing the names and icons were to do this, it would not be worth it because on net it would make the experience worse.

---

Fortunately, we have a better way to do this sort of thing. Take a look at the `Account` type in src/types.js, and the properties `zulipVersion` and `zulipFeatureLevel` there. Each of those is data from the server, which we keep around for all of the user's accounts. If you search the codebase for `zulipFeatureLevel` and look at the code around the results, particularly in accountsReducer.js and migrations.js, you'll see how we maintain that data.

We can store the `realm_icon` and `realm_name` information for each account in the same way as we store its `zulip_feature_level`. Then we'll have that data immediately when the user looks at this screen, without having to wait for a network request.

When we do, there will still sometimes be accounts that we don't have the data for -- in particular, when the user first upgrades to a version of the app that has this feature, that will be true at first for any existing accounts they're logged into, until the next time they use that account. This screen needs to handle that case gracefully -- in particular, it should not make the user wait in that case either.

A good-enough solution would be that we just leave the icon and the name blank, and show the URL and email like we currently do.